client/src/components/ActiveChat/Messages.js

```diff
@@ -3,18 +3,32 @@ import { Box } from "@material-ui/core";
 import { SenderBubble, OtherUserBubble } from "../ActiveChat";
 import moment from "moment";
 
+const findLastReadMessageId = (messages, userId) => {
+  let lastReadMessageId = -1;
+  messages.forEach(message=>{
+    if (message.id > lastReadMessageId && message.read === true && message.senderId === userId)
+      {
+        lastReadMessageId = message.id
+      }
+  })
+  return lastReadMessageId === -1 ? undefined: lastReadMessageId;
+}
```

Comment:
Nice job extracting this logic into a external function; however, this function is run on every render. It would be more efficient if this could be placed inside Redux reducer.

Reply:
Fixed